sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2395,6 +2421,70 @@ class Analyzer(override val catalogManager: CatalogManager)
     }
   }
 
+  /**
+   * This rule resolves lateral joins.
+   */
+  object ResolveLateralJoin extends Rule[LogicalPlan] {
```

Comment:
`ResolveReferencesInLateralJoin`?

Reply:
Q: Btw, is it difficult to merge this resolution logic with the `ResolveReferences` one?